### PR TITLE
Download Maven from apache.org archives [skip ci]

### LINKED ISSUE #10224: [BUG] DBR builds fails when installing Maven
**Describe the bug**

calling `./jenkins/databricks/build.sh`
```
E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/universe/g/guice/libguice-java_4.2.1-1ubuntu0.1_all.deb  404  Not Found [IP: 91.189.91.82 80]
```
**Steps/Code to reproduce bug**

Minimum repro
```
sudo apt install maven
```

We can download from Maven from https://archive.apache.org/dist/maven/maven-3/3.6.3/binaries/

### PR DESCRIPTION
**Pull request by gerashegalov**
Fixes #10224 

Replace broken install using apt by downloading Maven from apache.org.

Signed-off-by: Gera Shegalov <gera@apache.org>

### CODE REVIEW

**Comment by gerashegalov** [comment]
build

File: jenkins/databricks/build.sh [hunk 1]
```diff
@@ -52,7 +52,13 @@ declare -A artifacts
 initialize()
 {
     # install rsync to be used for copying onto the databricks nodes
-    sudo apt install -y maven rsync
+    sudo apt install -y rsync
+
+    if [[ ! -d $HOME/apache-maven-3.6.3 ]]; then
+        wget https://archive.apache.org/dist/maven/maven-3/3.6.3/binaries/apache-maven-3.6.3-bin.tar.gz -P /tmp
+        tar xf /tmp/apache-maven-3.6.3-bin.tar.gz -C $HOME
+        sudo ln -s $HOME/apache-maven-3.6.3/bin/mvn /usr/local/bin/mvn
```
Comment (jlowe) [inline jenkins/databricks/build.sh:60]: Nit: Should the .tar.gz be deleted here?

**Review by jlowe — APPROVED** [review]
Small nit that's not mustfix.

**Comment by gerashegalov** [comment]
Looking into 330db build failure on Databricks Azure 11.3

**Comment by gerashegalov** [comment]
Rerunning with skip-ci because 330db will fail due to #10228 and non-Databricks CI is not affected by this change

**Comment by gerashegalov** [comment]
build

**Pull request merged**